simpleclient_pushgateway/src/main/java/io/prometheus/client/exporter/PushGateway.java

```diff
@@ -235,7 +264,8 @@ public void delete(String job, String instance) throws IOException {
   }
 
   void doRequest(CollectorRegistry registry, String job, Map<String, String> groupingKey, String method) throws IOException {
-    String url = "http://" + address + "/metrics/job/" + URLEncoder.encode(job, "UTF-8");
+    String url = gatewayBaseURL + URLEncoder.encode(job, "UTF-8");
```

Comment:
the `gatewayBaseURL` is now always `<scheme>://<host>[:<port>]/metrics/job/`, so we can just concatenate the job name and grouping keys.